lib/warden/jwt_auth/strategy.rb

```diff
@@ -25,6 +29,13 @@ def authenticate!
 
       private
 
+      def path_is_dispatch_request_path?
+        current_path = EnvHelper.path_info(env)
+        dispatch_requests.any? do |tuple|
+          current_path.match(tuple.last)
+        end
+      end
```

Comment:
Should we also check the request method?

Reply:
Yeah you're quite right here, if a user visits a dispatch request path via a `GET` for example, we want the strategy to actually run and indicate to devise that they are already logged in and trigger a redirect without inserting a fresh token into the headers. I will tweak my code so we also check the request method like it is done inside the hooks.